core/unit_test/CMakeLists.txt

```diff
@@ -453,6 +453,7 @@ if(Kokkos_ENABLE_THREADS)
     UnitTest_Threads
     SOURCES ${Threads_SOURCES}
     UnitTestMainInit.cpp
+    threads/TestThreads_MDTeamParallelism.cpp
```

Comment:
We don't want to add that test manually for every backend. You should add it in the glob and remove it for the backend that are not supported.

Reply:
Initially, this was done this way so that each backend PR wouldn't need to know which other backend has already been merged, but I see what you are suggesting would've work better.
Current plan is to open another PR after all backend impls are merged to clean up these unnecessary additions per backend in CMakeLists. At that point, this test will only be in the list of the common tests for all backends.